MaterialSkin/Controls/MaterialExpansionPanel.cs

```diff
@@ -0,0 +1,607 @@
+                {
+                    if (_collapse)
+                    {
+                        //Draw Expand button
+                        System.Drawing.Drawing2D.GraphicsPath pth = new System.Drawing.Drawing2D.GraphicsPath();
+                        PointF TopLeft = new PointF(_expandcollapseBounds.X + 6, _expandcollapseBounds.Y + 9);
+                        PointF MidBottom = new PointF(_expandcollapseBounds.X + 12, _expandcollapseBounds.Y + 15);
+                        PointF TopRight = new PointF(_expandcollapseBounds.X + 18, _expandcollapseBounds.Y + 9);
+                        pth.AddLine(TopLeft, MidBottom);
+                        pth.AddLine(TopRight, MidBottom);
+                        g.DrawPath(formButtonsPen, pth);
+                    }
+                    else
+                    {
+                        // Draw Collapse button
+                        System.Drawing.Drawing2D.GraphicsPath pth = new System.Drawing.Drawing2D.GraphicsPath();
+                        PointF BottomLeft = new PointF(_expandcollapseBounds.X + 6, _expandcollapseBounds.Y + 15);
+                        PointF MidTop = new PointF(_expandcollapseBounds.X + 12, _expandcollapseBounds.Y + 9);
+                        PointF BottomRight = new PointF(_expandcollapseBounds.X + 18, _expandcollapseBounds.Y + 15);
+                        pth.AddLine(BottomLeft, MidTop);
+                        pth.AddLine(BottomRight, MidTop);
+                        g.DrawPath(formButtonsPen, pth);
+                    }
+                }
+            }
+
+            if (!_collapse && _showValidationButtons)
+            {
+                //Draw divider
+                g.DrawLine(new Pen(SkinManager.DividersColor, 1), new Point(0, Height - _footerHeight), new Point(Width, Height - _footerHeight));
+
+                //Validation Button
+                using (NativeTextRenderer NativeText = new NativeTextRenderer(g))
+                {
+                    Point _textLocation = new Point(_savebuttonBounds.X + 8, _savebuttonBounds.Y);
+
+                    NativeText.DrawTransparentText(
+                    ValidationButtonText,
+                    SkinManager.getLogFontByType(MaterialSkinManager.fontType.Body1),
+                    Enabled && _savebuttonEnable ? (_buttonState == ButtonState.SaveOver) ? SkinManager.ColorScheme.AccentColor : SkinManager.TextHighEmphasisColor : SkinManager.TextDisabledOrHintColor,
+                    _textLocation,
+                    _savebuttonBounds.Size,
+                    NativeTextRenderer.TextAlignFlags.Left | NativeTextRenderer.TextAlignFlags.Middle);
+                }
+
+                //Cancel Button
+                using (NativeTextRenderer NativeText = new NativeTextRenderer(g))
+                {
+                    Point _textLocation = new Point(_cancelbuttonBounds.X + 8, _cancelbuttonBounds.Y);
```

Comment:
Same as validation, but this one should also consider the validation button text length-size

Reply:
Right, _cancelbuttonBounds update was missing.